platforms/cuda/src/CudaKernels.cpp

```diff
@@ -2045,8 +2185,80 @@ double CudaCalcNonbondedForceKernel::execute(ContextImpl& context, bool includeF
             cuEventRecord(pmeSyncEvent, pmeStream);
             cu.restoreDefaultStream();
         }
+
+        // As written, we check only the Electrostatic grid pointer to get here.  We could separate them out, but for
+        // now we assume that LJPME can only be used if electrostatic PME is also active.
+        if(doLJPME){
+            if (usePmeStream)
+                cu.setCurrentStream(dispersionPmeStream);
```

Comment:
Ok, so I completely agree about recomputing the C6 coefficients in the real space code.  For the reciprocal space, I have to pass data [here](https://github.com/pandegroup/openmm/pull/1557/files#diff-6647053ad41d2b5cc13dce3d49c64b1aR25) and [here](https://github.com/pandegroup/openmm/pull/1557/files#diff-6647053ad41d2b5cc13dce3d49c64b1aR244).  My first pass was to make a new C6 array, but now I think it'd be much better to group the C6 and xyz coords together in a posq-like array instead.  Passing in xyz, sigma and epsilon separately makes no sense to me.  Passing C6 as a distinct array is far less cache efficient than bundling with xyz, as is done for posq.  Agree?  Btw, I granted write permission to my repo in case you want to mess around with any of this stuff.  I'm happy to make the changes, but wanted to give you an easy route to tweaking things if you'd rather do that too.